
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
        <link rel="canonical" href="http://sepsis-omics.github.io/tutorials/modules/spades/">
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <link rel="shortcut icon" href="../../favicon.ico">
      
      <meta name="generator" content="mkdocs-0.17.2, mkdocs-material-1.6.1">
    
    
      
        <title>Genome assembly with Spades - ABRPI-Training</title>
      
    
    
      <script src="../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application-4d0d3f2fbf.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
  
  
  
    <body>
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="http://sepsis-omics.github.io/tutorials/" title="ABRPI-Training" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Genomics
                </span>
              
            
            Genome assembly with Spades
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
            


  


  <a href="https://github.com/sepsis-omics/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../workshop_overview/" title="Genomics" class="md-tabs__link md-tabs__link--active">
          Genomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../dge/" title="Transcriptomics" class="md-tabs__link">
          Transcriptomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xtandem/" title="Proteomics" class="md-tabs__link">
          Proteomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xcms/" title="Metabolomics" class="md-tabs__link">
          Metabolomics
        </a>
      
    </li>
  

      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    ABRPI-Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/sepsis-omics/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../workshop_overview/" title="Genomics workshop overview" class="md-nav__link">
      Genomics workshop overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pear/" title="Merge reads with Pear" class="md-nav__link">
      Merge reads with Pear
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../velvet/" title="Genome assembly with Velvet" class="md-nav__link">
      Genome assembly with Velvet
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Genome assembly with Spades
      </label>
    
    <a href="./" title="Genome assembly with Spades" class="md-nav__link md-nav__link--active">
      Genome assembly with Spades
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-and-view-data" title="Import and view data" class="md-nav__link">
    Import and view data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#galaxy" title="Galaxy" class="md-nav__link">
    Galaxy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-data" title="The data" class="md-nav__link">
    The data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-reads-with-spades" title="Assemble reads with Spades" class="md-nav__link">
    Assemble reads with Spades
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-the-output" title="Examine the output" class="md-nav__link">
    Examine the output
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant finding" class="md-nav__link">
      Variant finding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../access_data/" title="Public data &rarr; assembly, annotation" class="md-nav__link">
      Public data &rarr; assembly, annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../abricate/" title="Finding antibiotic-resistant genes" class="md-nav__link">
      Finding antibiotic-resistant genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../viral_genomes/" title="Viral genome sequencing" class="md-nav__link">
      Viral genome sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows" class="md-nav__link">
      Galaxy workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../frogs/" title="Metagenomics with FROGS" class="md-nav__link">
      Metagenomics with FROGS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification with Kraken" class="md-nav__link">
      Sample identification with Kraken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pacbio/" title="PacBio assembly with SMRT portal" class="md-nav__link">
      PacBio assembly with SMRT portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cmdline_assembly/" title="PacBio assembly with command line tools" class="md-nav__link">
      PacBio assembly with command line tools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades_cmdline/" title="Spades assembly - commandline" class="md-nav__link">
      Spades assembly - commandline
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mauve/" title="Genome alignment with Mauve" class="md-nav__link">
      Genome alignment with Mauve
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary/" title="Pangenomes with Roary/Phandango - command line" class="md-nav__link">
      Pangenomes with Roary/Phandango - command line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary-galaxy/" title="Pangenomes with Roary/Phandango - Galaxy" class="md-nav__link">
      Pangenomes with Roary/Phandango - Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cmdline_assembly_v2/" title="Long read assembly workshop" class="md-nav__link">
      Long read assembly workshop
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../dge/" title="Differential gene expression using Galaxy and Degust" class="md-nav__link">
      Differential gene expression using Galaxy and Degust
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kallisto/" title="Differential gene expression using Kallisto and Degust" class="md-nav__link">
      Differential gene expression using Kallisto and Degust
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xtandem/" title="Protein Identification" class="md-nav__link">
      Protein Identification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pathway_tools/annotation/" title="Pathway annotation and prediction" class="md-nav__link">
      Pathway annotation and prediction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Metabolomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metabolomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xcms/" title="Metabolite identification" class="md-nav__link">
      Metabolite identification
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-and-view-data" title="Import and view data" class="md-nav__link">
    Import and view data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#galaxy" title="Galaxy" class="md-nav__link">
    Galaxy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-data" title="The data" class="md-nav__link">
    The data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-reads-with-spades" title="Assemble reads with Spades" class="md-nav__link">
    Assemble reads with Spades
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-the-output" title="Examine the output" class="md-nav__link">
    Examine the output
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/sepsis-omics/tutorials/edit/master/docs/modules/spades/index.md" title="Edit this page" class="md-icon md-content__icon">edit</a>
                
                
                <p><br></p>
<h1 id="assembly-using-spades">Assembly using Spades</h1>
<p>Keywords: de novo assembly, Spades, Galaxy, Microbial Genomics Virtual Lab</p>
<h2 id="background">Background</h2>
<p>Spades is one of a number of <em>de novo</em> assemblers that use short read sets as input (e.g. Illumina Reads), and the assembly method is based on de Bruijn graphs. For information about Spades see this <a href="http://bioinf.spbau.ru/spades">link</a>.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>assemble the reads using Spades, and</li>
<li>examine the output assembly.</li>
</ol>
<h2 id="import-and-view-data">Import and view data</h2>
<!-- If you have completed the previous tutorial on [Quality Control](/modules/fastqc/index.md), you should already have the required files in your current Galaxy history. If not, see how to get them [here](/modules/galaxy/index.md).
-->

<h3 id="galaxy">Galaxy</h3>
<p>If you are using Galaxy-Mel or Galaxy-Qld, import the files:</p>
<ul>
<li>In your browser, go to <a href="http://galaxy-mel.genome.edu.au/galaxy/">Galaxy-Mel</a> or <a href="https://galaxy-qld.genome.edu.au/galaxy">Galaxy-Qld</a> </li>
<li>In the top Galaxy panel, go to <ss>User</ss> and log in (or register, and then log in)</li>
<li>In the top Galaxy panel, go to <ss>Shared Data</ss> and click on the drop down arrow</li>
<li>Click on <ss>Histories</fn></li>
<li>Click on <fn>Genomics-workshop</fn> and then (over in the top right) <ss>Import history</ss></li>
<li>The files will now be listed in the right hand panel (your current history).</li>
</ul>
<p>(Alternatively, see <a href="../galaxy/">here</a> for information about how to start with Galaxy, and <a href="../data-dna/">here</a> for the link to import the Galaxy history for this tutorial, if you don&rsquo;t already have them in your history.)</p>
<h3 id="the-data">The data</h3>
<p>The read set for today is from an imaginary <em>Staphylococcus aureus</em> bacterium with a miniature genome.</p>
<ul>
<li>
<p>The whole genome shotgun method used to sequence our mutant strain read set was produced on an Illumina DNA sequencing instrument.</p>
</li>
<li>
<p>The files we need for assembly are the <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</p>
</li>
<li>
<p>(We don&rsquo;t need the reference genome sequences for this tutorial).</p>
</li>
<li>
<p>The reads are paired-end.</p>
</li>
<li>
<p>Each read is 150 bases long. <!--(before trimming)--></p>
</li>
<li>
<p>The number of bases sequenced is equivalent to 19x the genome sequence of the wildtype strain. (Read coverage 19x - rather low!).</p>
</li>
</ul>
<!--
- <fn>wildtype.fna</fn>: the reference genome sequence of the wildtype strain in fasta format (a header line, then the nucleotide sequence of the genome)

- <fn>wildtype.gff</fn>: the reference genome sequence of the wildtype strain in general feature format (a list of features - one feature per line, then the nucleotide sequence of the genome).

- <fn>wildtype.gbk</fn>: the reference genome sequence in genbank format.
-->

<ul>
<li>Click on the View Data button (the <img alt="Eye icon" src="./images/image04.png" />) next to one of the FASTQ sequence files.</li>
</ul>
<!--
- The gff file should look like this:
- Brief Discussion about the GFF format (FIXME: add)
![GFF format](images/image08.png)

## Evaluate the input reads

Questions you might ask about your input reads include:

- How good is my read set?
- Do I need to ask for a new sequencing run?  
- Is it suitable for the analysis I need to do?

We will evaluate the input reads using the FastQC tool.

- This runs a standard series of tests on your read set and returns a relatively easy-to-interpret report.
- We will use the FastQC tool in Galaxy to evaluate the quality of one of our FASTQ files.
- Go to <ss>Tools &rarr; NGS:Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss>
- Select <fn>mutant_R1.fastq</fn>
- <ss>Execute</ss>
- Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint:![Eye icon](./images/image04.png)). It has a summary at the top of
the page and a number of graphs.

Some of the important outputs of FastQC for our purposes are:

-   <ss>Basic Statistics: Sequence length</ss>: will be important in setting maximum k-mer size value for assembly
-   <ss>Basic Statistics: Encoding</ss>: Quality encoding type: important for quality trimming software
-   <ss>Basic Statistics: % GC</ss>: high GC organisms don’t tend to assemble well and may have an uneven read coverage distribution.
-   <ss>Basic Statistics: Total sequences</ss>: Total number of reads: gives you an idea of coverage.
-   <ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run.
-   <ss>Per base N content</ss>: Presence of large numbers of Ns in reads: may point to poor quality sequencing run. You would need to trim these reads to remove Ns.
-   <ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes or adapter sequences.

Although we have warnings for two outputs (per base sequence content; Kmer content), we can ignore these for now. For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of ["good"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and ["bad"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data. We won’t be doing anything to these data to clean it up as there isn’t much need. Therefore we will get on with the assembly!

-->

<h2 id="assemble-reads-with-spades">Assemble reads with Spades</h2>
<ul>
<li>
<p>We will perform a <em>de novo</em> assembly of the mutant FASTQ reads into long contiguous sequences (in FASTA format.)</p>
</li>
<li>
<p>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; spades</ss></p>
</li>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>Run only Assembly</ss>: <em>Yes</em> [the <em>Yes</em> button should be darker grey]</li>
<li><ss>Kmers to use separated by commas:</ss> <em>33,55,91</em>  [note: no spaces]  </li>
<li><ss>Coverage cutoff:</ss> <em>auto</em>  </li>
<li><ss>Files &rarr; Forward reads:</ss> <fn>mutant_R1.fastq</fn>  </li>
<li><ss>Files &rarr; Reverse reads:</ss> <fn>mutant_R2.fastq</fn>  </li>
</ul>
</li>
<li>
<p>Your tool interface should look like this:</p>
</li>
</ul>
<p><img alt="Spades interface" src="./images/image03.png" /></p>
<ul>
<li>Click <ss>Execute</ss></li>
</ul>
<h2 id="examine-the-output">Examine the output</h2>
<ul>
<li>Galaxy is now running Spades on the reads for you.</li>
<li>
<p>When it is finished, you will have five (or more) new files in your history, including:</p>
<ul>
<li>two FASTA files of the resulting contigs and scaffolds</li>
<li>two files for statistics about these</li>
<li>the Spades logfile</li>
</ul>
</li>
</ul>
<p><img alt="spades output" src="./images/output_files.png" /></p>
<ul>
<li>Click on the View Data button <img alt="Eye icon" src="./images/image04.png" /> on each of the files.</li>
<li>Note that the short reads have been assembled into much longer contigs.</li>
<li>(However, in this case, the contigs have not been assembled into larger scaffolds.)</li>
<li>The stats files will give you the length of each of the contigs, and the file should look something like this:</li>
</ul>
<p><img alt="spades output contigs" src="./images/contig_stats.png" /></p>
<!-- ## What next?

- [Annotate the genome using Prokka.](/modules/prokka/index.md)
-->
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../velvet/" title="Genome assembly with Velvet" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Genome assembly with Velvet
              </span>
            </div>
          </a>
        
        
          <a href="../prokka/" title="Genome annotation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Genome annotation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2016-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://omics.data.edu.au/" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application-6b599127bc.js"></script>
      <script>app.initialize({url:{base:"../.."}})</script>
      
        <script src="../../search/require.js"></script>
      
        <script src="../../search/search.js"></script>
      
    
    
      
      <script>!function(e,t,a,n,o,c,i){e.GoogleAnalyticsObject=o,e[o]=e[o]||function(){(e[o].q=e[o].q||[]).push(arguments)},e[o].l=1*new Date,c=t.createElement(a),i=t.getElementsByTagName(a)[0],c.async=1,c.src=n,i.parentNode.insertBefore(c,i)}(window,document,"script","https://www.google-analytics.com/analytics.js","ga"),ga("create","UA-21794901-5","www.melbournebioinformatics.org.au/tutorials/"),ga("set","anonymizeIp",!0),ga("send","pageview");var links=document.getElementsByTagName("a");Array.prototype.map.call(links,function(e){e.host!=document.location.host&&e.addEventListener("click",function(){var t=e.getAttribute("data-md-action")||"follow";ga("send","event","outbound",t,e.href)})});var query=document.forms.search.query;query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})</script>
      
    
  </body>
</html>