
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <link rel="canonical" href="http://sepsis-omics.github.io/tutorials/modules/scripts/">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.26">
    
    
      
        <title>Index - ABRPI-Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.6543a935.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bash-scripts" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="ABRPI-Training" class="md-header__button md-logo" aria-label="ABRPI-Training" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ABRPI-Training
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Index
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sepsis-omics/tutorials
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="ABRPI-Training" class="md-nav__button md-logo" aria-label="ABRPI-Training" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    ABRPI-Training
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sepsis-omics/tutorials
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Genomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../workshop_overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomics workshop overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../galaxy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Starting with Galaxy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../data-dna/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Training dataset
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../fastqc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quality control
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pear/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Merge reads with Pear
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../velvet/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome assembly with Velvet
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../spades/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome assembly with Spades
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../prokka/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../snippy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Variant finding
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../access_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Public data &rarr; assembly, annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../abricate/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Finding antibiotic-resistant genes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../viral_genomes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Viral genome sequencing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../workflows/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Galaxy workflows
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../frogs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Metagenomics with FROGS
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../kraken/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sample identification with Kraken
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pacbio/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PacBio assembly with SMRT portal
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cmdline_assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    PacBio assembly with command line tools
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../spades_cmdline/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spades assembly - commandline
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mauve/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genome alignment with Mauve
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../roary/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pangenomes with Roary/Phandango - command line
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../roary-galaxy/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pangenomes with Roary/Phandango - Galaxy
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cmdline_assembly_v2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Long read assembly workshop
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Transcriptomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Transcriptomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../dge/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Differential gene expression using Galaxy and Degust
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../kallisto/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Differential gene expression using Kallisto and Degust
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Proteomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Proteomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../xtandem/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Protein Identification
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pathway_tools/annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pathway annotation and prediction
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Metabolomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Metabolomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../xcms/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Metabolite identification
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../about/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    About
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prepare-the-bash-script" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare the bash script
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#edit-the-bash-script" class="md-nav__link">
    <span class="md-ellipsis">
      Edit the bash script
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run" class="md-nav__link">
    <span class="md-ellipsis">
      Run
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<p><br></p>
<h1 id="bash-scripts">Bash scripts</h1>
<p>It is possible to put some commands into a script, to run automatically.</p>
<p>Here, we will put commands for several tools into a bash script and use it to assemble a bacterial genome.</p>
<p>Bash is a programming language, and we can run a bash script in the bash shell (e.g. the terminal).</p>
<h2 id="prepare-the-bash-script">Prepare the bash script</h2>
<ul>
<li>Open the terminal.</li>
<li>Open a text editor, e.g., nano.</li>
</ul>
<div class="codehilite"><pre><span></span><code>nano<span class="w"> </span>myscript.sh
</code></pre></div>

<ul>
<li>We now have a script called myscript.sh</li>
<li>In the script, write</li>
</ul>
<div class="codehilite"><pre><span></span><code><span class="ch">#!/bin/bash  </span>
<span class="nb">echo</span><span class="w"> </span>This<span class="w"> </span>is<span class="w"> </span>my<span class="w"> </span>assembly<span class="w"> </span>script
</code></pre></div>

<ul>
<li>
<p>Ctrl-X to exit nano.</p>
</li>
<li>
<p>Make the script executable:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span><code>chmod<span class="w"> </span><span class="m">755</span><span class="w"> </span>myscript.sh
</code></pre></div>

<ul>
<li>Run the script to test.</li>
</ul>
<div class="codehilite"><pre><span></span><code>./myscript.sh
</code></pre></div>

<ul>
<li>The text should print to the screen:</li>
</ul>
<div class="codehilite"><pre><span></span><code>This<span class="w"> </span>is<span class="w"> </span>my<span class="w"> </span>assembly<span class="w"> </span>script
</code></pre></div>

<h2 id="edit-the-bash-script">Edit the bash script</h2>
<ul>
<li>Now we will add to the bash script.</li>
<li>This will be a simple script that is designed to assemble, trim, and correct.</li>
<li>It will likely only work in simple cases.</li>
<li>Open nano</li>
</ul>
<div class="codehilite"><pre><span></span><code>nano<span class="w"> </span>myscript.sh
</code></pre></div>

<ul>
<li>and type in the following
(#comments and echo statements are optional but probably useful):</li>
</ul>
<div class="codehilite"><pre><span></span><code><span class="ch">#!/bin/bash</span>
<span class="c1">#a script for assembling a bacterial genome</span>

<span class="c1">#set a variable for the number of CPUS</span>
<span class="c1">#this can then be easily changed if needed</span>
<span class="nv">CPUS</span><span class="o">=</span><span class="m">16</span>

<span class="c1">#this script requires 5 arguments</span>
<span class="c1">#we will test whether 5 were entered</span>
<span class="k">if</span><span class="w"> </span><span class="o">[</span><span class="w"> </span>!<span class="w"> </span><span class="nv">$#</span><span class="w"> </span>-eq<span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="o">]</span><span class="w"> </span><span class="c1">#if there are not 5 arguments</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;ERROR: needs 5 arguments&quot;</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;usage: </span><span class="nv">$0</span><span class="s2"> outdir genomeSize (e.g. 2.1m) pacbio.fq R1.fq R2.fq&quot;</span>
<span class="w">    </span><span class="c1">#the usage shows what arguments are required</span>
<span class="w">    </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="c1">#exits</span>
<span class="k">fi</span>

<span class="c1">#report back the arguments entered:</span>
<span class="c1">#each argument is assigned a variable name from 1 to 5</span>
<span class="c1">#these variables are referred to with the $ sign</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your directory with output is called </span><span class="nv">$1</span><span class="s2">&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your genomeSize is </span><span class="nv">$2</span><span class="s2">&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your pacbio reads are in file </span><span class="nv">$3</span><span class="s2">&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your R1 reads are in file </span><span class="nv">$4</span><span class="s2">&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your R2 reads are in file </span><span class="nv">$5</span><span class="s2">&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="c1">#echo on its own puts in a blank line on the screen</span>

<span class="c1">#run canu to assemble the reads</span>
mkdir<span class="w"> </span>-p<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">&quot;</span><span class="w">  </span><span class="c1">#the p creates intermediate directories where required</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;now running canu&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your canu output is in directory &quot;</span><span class="nv">$1</span>/canu<span class="s2">&quot;&quot;</span>
nice<span class="w"> </span>canu<span class="w"> </span>-p<span class="w"> </span>canu<span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu&quot;</span><span class="w"> </span><span class="nv">genomeSize</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$2</span><span class="s2">&quot;</span><span class="w"> </span>-pacbio-raw<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$3</span><span class="s2">&quot;</span>
<span class="c1">#commands are preceeded by &#39;nice&#39; in this script for particular servers</span>

<span class="c1">#test canu output: number of contigs</span>
<span class="c1">#make a variable called num_contigs</span>
<span class="nv">num_contigs</span><span class="o">=</span><span class="k">$(</span>grep<span class="w"> </span>-c<span class="w"> </span><span class="s1">&#39;&gt;&#39;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.contigs.fasta&quot;</span><span class="k">)</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;number of contigs found by canu is: </span><span class="nv">$num_contigs</span><span class="s2">&quot;</span>
<span class="nb">echo</span>
<span class="k">if</span><span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="nv">$num_contigs</span><span class="w"> </span>-gt<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="o">]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;there are more than 10 contigs&quot;</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;the analysis is stopping&quot;</span>
<span class="w">    </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
<span class="c1">#in more complex scripts, you can add the option to repeat the analysis with different parameters, etc.</span>
<span class="k">fi</span>

<span class="c1">#run circlator to trim the contigs</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;now running circlator to trim and orient contigs&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your circlator output is in directory &quot;</span><span class="nv">$1</span>/circlator<span class="s2">&quot; &quot;</span>
nice<span class="w"> </span>circlator<span class="w"> </span>all<span class="w"> </span>--threads<span class="w"> </span><span class="nv">$CPUS</span><span class="w"> </span>--verbose<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.contigs.fasta&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.correctedReads.fasta.gz&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/circlator&quot;</span>
<span class="c1">#in more complex scripts, you can add information about which contigs were merged; where contigs were oriented; how contigs were trimmed, etc.</span>
<span class="c1">#rename the contigs file</span>
cp<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/circlator/06.fixstart.fasta&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your contigs from circlator are now in the file called contigs.fa&quot;</span>
<span class="c1">#note: in this case, these contigs may not have been trimmed.</span>

<span class="c1">#find smaller plasmids</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;now looking in illumina reads for small plasmids&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;indexing the contigs.fa file&quot;</span>
nice<span class="w"> </span>bwa<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;aligning illumina reads to the contigs.fa file&quot;</span>
nice<span class="w"> </span>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="nv">$CPUS</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$4</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$5</span><span class="s2">&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;indexing the aln.bam file&quot;</span>
nice<span class="w"> </span>samtools<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;extracting the unmapped illumina reads&quot;</span>
samtools<span class="w"> </span>fastq<span class="w"> </span>-f<span class="w"> </span><span class="m">4</span><span class="w"> </span>-1<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R1.fastq&quot;</span><span class="w"> </span>-2<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R2.fastq&quot;</span><span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.RS.fastq&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;assembling the unmapped reads with spades&quot;</span>
<span class="nb">echo</span>
nice<span class="w"> </span>spades.py<span class="w"> </span>-t<span class="w"> </span><span class="nv">$CPUS</span><span class="w"> </span>-1<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R1.fastq&quot;</span><span class="w"> </span>-2<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R2.fastq&quot;</span><span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.RS.fastq&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/spades_assembly&quot;</span>
<span class="nb">echo</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;your output is in the spades_assembly directory&quot;</span>
<span class="nb">echo</span>

<span class="c1">#check if any contigs are &gt; 1500bp</span>
seqtk<span class="w"> </span>seq<span class="w"> </span>-L<span class="w"> </span><span class="m">1500</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/spades_assembly/contigs.fasta&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/illumina_contigs.fasta&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;extracting contigs &gt; 1500bp into illumina_contigs.fasta&quot;</span>
<span class="nb">echo</span>
<span class="c1">#in more complex scripts, you can trim and orient these contigs, or do further investigation into whether they are true plasmids</span>

<span class="c1">#join the canu pacbio contigs and the small plasmids</span>
cat<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/illumina_contigs.fasta&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>

<span class="nb">echo</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;joining small contigs to the pacbio contigs as allcontigs.fasta&quot;</span>
<span class="nb">echo</span>

<span class="c1">#correct this assembly with Pilon</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;preparing to correct contigs with illumina reads, using Pilon&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;indexing allcontigs.fasta&quot;</span>
nice<span class="w"> </span>bwa<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;aligning illumina reads to allcontigs.fasta&quot;</span>
nice<span class="w"> </span>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="nv">$CPUS</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$4</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$5</span><span class="s2">&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;indexing aln.bam&quot;</span>
nice<span class="w"> </span>samtools<span class="w"> </span>index<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;indexing allcontigs.fasta&quot;</span>
nice<span class="w"> </span>samtools<span class="w"> </span>faidx<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;correcting with Pilon&quot;</span>
nice<span class="w"> </span>pilon<span class="w"> </span>--genome<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span><span class="w"> </span>--frags<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span><span class="w"> </span>--output<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/pilon1&quot;</span><span class="w"> </span>--fix<span class="w"> </span>all<span class="w"> </span>--mindepth<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>--changes<span class="w"> </span>--verbose<span class="w"> </span>--threads<span class="w"> </span><span class="nv">$CPUS</span>
<span class="nb">echo</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Pilon finished&quot;</span>
<span class="nb">echo</span>

<span class="c1">#count how many changes were made</span>
<span class="nv">pilon_changes</span><span class="o">=</span><span class="k">$(</span>wc<span class="w"> </span>-l<span class="w"> </span>&lt;<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/pilon1.changes&quot;</span><span class="k">)</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;there were </span><span class="nv">$pilon_changes</span><span class="s2"> corrections made&quot;</span>
<span class="nb">echo</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Assembly complete&quot;</span>
<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;The sequence is saved as &quot;</span><span class="nv">$1</span>/pilon1.fasta<span class="s2">&quot;&quot;</span>

<span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
</code></pre></div>

<h2 id="run">Run</h2>
<p>To run, type in</p>
<div class="codehilite"><pre><span></span><code>./myscript.sh<span class="w"> </span>outdir<span class="w"> </span><span class="m">1</span>.9m<span class="w"> </span>subreads.fastq<span class="w"> </span>R1.fastq<span class="w"> </span>R2.fastq
</code></pre></div>

<ul>
<li><strong>outdir</strong> is the name of the output directory</li>
<li><strong>1.9m</strong> is the approximate genome size for the species</li>
<li><strong>subreads.fastq</strong> is the name of the Pacbio reads file</li>
<li><strong>R1.fastq</strong> is the name of the R1 Illumina reads file</li>
<li>
<p><strong>R2.fastq</strong> is the name of the R2 Illumina reads file</p>
</li>
<li>
<p>If the Pacbio and Illumina reads files are not in the current directory, put in the whole path to those files</p>
</li>
<li>
<p>The script should run to the end or may exit early if necessary (e.g. the canu assembly found too many contigs).</p>
</li>
<li>
<p>There are many refinements and additions that could be made to this script, and some of these are noted as comments above.</p>
</li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2016-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.ad660dcc.min.js"></script>
      
    
  </body>
</html>