<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="../../favicon.ico">
        

	<title>Using Galaxy - Sepsis Tutorials</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Sepsis Tutorials</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../genomics/">Overview</a>
</li>

                    
                        
<li class="active">
    <a href="./">Using Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../../genomics/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../data-dna/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#using-galaxy">Using Galaxy</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#login-to-galaxy">Login to Galaxy</a></li>
        
            <li><a href="#put-data-onto-galaxy">Put data onto Galaxy</a></li>
        
            <li><a href="#galaxy-tools">Galaxy tools</a></li>
        
            <li><a href="#what-next">What next?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="using-galaxy">Using Galaxy</h1>
<h2 id="background">Background</h2>
<p>Galaxy is a web-based analysis and workflow platform designed for biologists to analyse their own data. It can be used to run a variety of bioinformatics tools. The selection of bioinformatics tools installed on the Galaxy instance we are using today caters for the analysis of bacterial genomics data sets.</p>
<p>Bioinformatics tools can be added from the Galaxy ‘tool shed’ or removed as necessary from your Galaxy instance.</p>
<p>Galaxy is an open, web-based platform. Details about the project can be found <a href="https://galaxyproject.org/">here</a>.</p>
<p>The Galaxy interface is separated into three parts. The <ss>Tools</ss> list on the left, the <ss>Viewing</ss> panel in the middle and the analysis and data <ss>History</ss> on the right. We will be looking at all three parts in this tutorial.</p>
<p><img alt="galaxy overview screenshot" src="./images/image05.png" /></p>
<p>This activity will familiarise you with the Galaxy interface. It will cover the following operations:</p>
<ul>
<li>Logging in to the server</li>
<li>Putting data onto Galaxy</li>
<li>Using some common tools</li>
</ul>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>Register and login to a Galaxy server.</li>
<li>Upload data to a Galaxy server from:<ul>
<li>A file on your local computer.</li>
<li>A file on a remote datastore with an accessible URL.  </li>
</ul>
</li>
<li>Use tools in Galaxy by:<ul>
<li>Accessing the tool via the tool menu.</li>
<li>Using the tool interface to run the particular tool.</li>
<li>Viewing/accessing the tool output.</li>
</ul>
</li>
</ol>
<h2 id="login-to-galaxy">Login to Galaxy</h2>
<p>Open a new tab or window on your web browser. Use Firefox or Chrome - please don’t use Internet Explorer or Safari.</p>
<p>In the address bar, type in the address of your galaxy server. Alternatively, you can access galaxy via the dashboard of your mGVL.</p>
<p><img alt="Galaxy URL" src="./images/image09.png" /></p>
<p>Click on <ss>User</ss> button on the right and either register or login.</p>
<p><img alt="Register or Login screenshot" src="./images/image04.png" /></p>
<p>If you haven&rsquo;t yet registered yet:</p>
<ul>
<li>Select: <ss>User &rarr; Register</ss></li>
<li>Enter your email, choose a password, and choose a user name.</li>
<li>Click <ss>Submit</ss></li>
</ul>
<p>If you have already registered:</p>
<ul>
<li>Select: <ss>User &rarr; Login</ss></li>
<li>Enter your username &amp; password.</li>
<li>Click <ss>Submit</ss></li>
</ul>
<h2 id="put-data-onto-galaxy">Put data onto Galaxy</h2>
<p>There are two main ways to put your data onto Galaxy; this section will run through both ways. First, we need to make a new history.</p>
<h3 id="make-a-new-history">Make a new history</h3>
<p>First, make a new folder to store the work we are about to perform.</p>
<ul>
<li>Click on the history menu button <img alt="history icon" src="./images/image02.png" /> at the top of the <ss>History</ss> panel.</li>
<li>Select <ss>Create New</ss></li>
<li>Click on <ss>Unnamed history</ss> to rename. Type in a new name.</li>
</ul>
<h3 id="datatypes">Datatypes</h3>
<p>What sort of file is being uploaded?</p>
<p>We need to tell Galaxy what sort of file is being uploaded. Some common datatypes (file formats) are: text, FASTA, FASTQ, VCF, GFF, GBK, and tabular.</p>
<h3 id="upload-a-file-from-your-own-computer">Upload a file from your own computer</h3>
<p>With this method you can get most of the files on your own computer into Galaxy.</p>
<h4 id="first-download-the-following-file-to-your-computer">First, download the following file to your computer:</h4>
<ul>
<li>Copy this URL and paste it into the address bar in your web browser: <a href="https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz">https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz</a><br />
This is a file stored on the cloud. Press <ss>Enter</ss>, and the file will download. Note the download location.</li>
</ul>
<p>&nbsp;</p>
<p><img alt="Browser address bar" src="./images/image06.png" /></p>
<p>&nbsp;</p>
<h4 id="next-upload-that-file-to-galaxy">Next, upload that file to Galaxy</h4>
<ul>
<li>From the Galaxy tool panel, click on <ss>Get Data &rarr; Upload File</ss>  </li>
<li>Click the <ss>Choose local file</ss> button  </li>
<li>Find and select the <fn>Contig_stats.txt.gz</fn> file you downloaded and click <ss>Open</ss>  </li>
<li>Set the <ss>Type</ss> to <em>tabular</em>  </li>
<li>Click the <ss>Start</ss> button  </li>
<li>Once the progress bar reaches 100%, click the <ss>Close</ss> button  </li>
<li>The file will now upload to your current history.</li>
</ul>
<h3 id="upload-a-file-from-a-url">Upload a file from a URL</h3>
<p>If a file exists on a web resource somewhere and you know its URL (Unique Resource Location - a web address) you can directly load it into Galaxy.</p>
<ul>
<li>From the tool panel, click on <ss>Get Data &rarr; Upload File</ss></li>
<li>Click on the <ss>Paste/Fetch Data</ss> button</li>
<li>Copy and paste the following web address into the URL/Text box:
<a href="https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz">https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz</a></li>
<li>Set the <ss>Type</ss> to <em>fastqsanger</em> (CAREFUL: not fastqCsanger)</li>
<li>Click <ss>Start</ss></li>
<li>Once the progress bar has reached 100%, click <ss>Close</ss>.</li>
<li>Note that Galaxy is smart enough to recognize that this is a compressed file and so it will uncompress it as it loads it.</li>
</ul>
<h3 id="upload-another-file-from-a-url">Upload another file from a URL</h3>
<p>Now we are going to upload another file from the remote data source.</p>
<ul>
<li>Repeat the above for: https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/MRSA0252.fna</li>
<li>Note: this file <ss>Type</ss> is <em>FASTA</em>.</li>
<li>The DNA sequence of <em>Staphylococcus aureus MRSA252</em> will be loaded into your history as a FASTA file.</li>
<li>Your <ss>History</ss> should now look like this:</li>
</ul>
<p><img alt="History screenshot" src="./images/image03.png" /></p>
<h3 id="the-data">The data</h3>
<p>A brief description of each of the three files uploaded to Galaxy:  </p>
<p><fn>Contigs_stats.txt</fn>  </p>
<ul>
<li>this file contains a table of summary data from a <em>de novo</em> genome assembly (the process of attempting to recover the full genome of an organism from the short read sequences produced by most DNA sequencing machines).</li>
<li>The columns contain a lot of information but the ones we will be using indicate the amount of data (or coverage) that went into making up each piece of the final assembly.</li>
</ul>
<p><fn>bacterial_std_err_1.fastq.gz</fn>  </p>
<ul>
<li>This file contains sequence reads, in the format produced by Illumina sequencing machines. Read more about the
<a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> format at Wikipedia.</li>
</ul>
<p><fn>MRSA0252.fna</fn></p>
<ul>
<li>This file contains the genome sequence of <em>Staphylococcus aureus MRSA252</em>. Read more about the <a href="https://en.wikipedia.org/wiki/FASTA_format">FASTA</a> format at Wikipedia.</li>
</ul>
<h2 id="galaxy-tools">Galaxy tools</h2>
<p>The purpose of this section is to help you become familiar with the way
tools are run on Galaxy.</p>
<p>We will see how to:</p>
<ul>
<li>rename files</li>
<li>summarize assembly statistics</li>
<li>convert file formats, and</li>
<li>find features in a DNA sequence.</li>
</ul>
<h3 id="rename-files">Rename files</h3>
<p>Two of the files in the <ss>History</ss> have very long and confusing names. File names can be changed by taking the following steps:</p>
<ul>
<li>Click on the edit icon <img alt="edit icon" src="./images/image07.png" /> next to the file in the <ss>History</ss> called: <fn>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq</fn></li>
<li>In the <ss>Name</ss> text box, give it a new name. Rename it to: <fn>typical.fastq</fn></li>
<li>Click the <ss>Save</ss> button.</li>
</ul>
<p>Repeat the process with another file:</p>
<ul>
<li>Find the file called: <fn>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/MRSA0252.fna</fn>  </li>
<li>Rename it to <fn>MRSA252.fna</fn></li>
</ul>
<p>Much better. There is other functionality hidden behind that edit icon <img alt="edit icon" src="./images/image07.png" /></p>
<p>You can change a file’s datatype, convert its format and many other things. Feel free to play around with them at a later date.</p>
<h3 id="summarize-assembly-statistics">Summarize assembly statistics</h3>
<p>We are going to produce a histogram of contig read-coverage depths and calculate the summary statistics from the <fn>Contig_stats.txt</fn> file.</p>
<p>To do this we need to make some changes to the <fn>Contig_stats.txt</fn> file:</p>
<ul>
<li>cut out a couple of columns from the file</li>
<li>remove a line from the file</li>
<li>produce a histogram</li>
</ul>
<h4 id="cut-out-and-keep-two-columns">Cut out and keep two columns</h4>
<ul>
<li>Click on the eye icon <img alt="eye icon" src="./images/image00.png" /> of the <fn>Contig_stats.txt</fn> file to have a look at it.  </li>
<li>Note that there are 18 columns in this file. We want column 1 and column 6.</li>
<li>Go to <ss>Tools &rarr; Text Manipulation &rarr; Cut</ss> and set the following:</li>
<li>Set <ss>Cut Columns</ss> to: <em>c1,c6</em></li>
<li><ss>Delimited by</ss>: <em>Tab</em></li>
<li><ss>From</ss>: <fn>Contig_stats.txt</fn></li>
<li>Click <ss>Execute</ss></li>
<li>Examine the new file by clicking on its eye icon <img alt="eye icon" src="./images/image00.png" />. We now have 2 columns instead of the 18 in the original file.</li>
</ul>
<h4 id="remove-the-header-lines-of-the-new-file">Remove the Header lines of the new file</h4>
<ul>
<li>Go to <ss>Tools &rarr; Text Manipulation &rarr; Remove beginning</ss> and set the following:</li>
<li><ss>Remove First</ss>: <em>1</em></li>
<li><ss>from</ss>: <fn>Cut on data1</fn></li>
<li>click <ss>Execute</ss></li>
<li>Note the the new file is the same as the previous one without the header line.</li>
</ul>
<h4 id="make-a-histogram">Make a histogram</h4>
<ul>
<li>Go to <ss>Tools &rarr; Graph/Display Data &rarr; Histogram</ss> and set the following:</li>
<li><ss>Dataset</ss>: <fn>Remove beginning on Data 4</fn></li>
<li><ss>Numerical column for X axis</ss> Column: 2</li>
<li><ss>Number of breaks</ss>: <em>25</em></li>
<li><ss>Plot title</ss>: <em>Histogram of Contig Coverage</em></li>
<li><ss>Label for X axis</ss>: <em>Coverage depth</em></li>
<li>Click <ss>Execute</ss></li>
<li>Click on the eye icon <img alt="eye icon" src="./images/image00.png" /> of the histogram to have a look at it. Note there are a few peaks. Maybe these correspond to single, double and triple copy number of these contigs.</li>
</ul>
<h4 id="calculate-summary-statistics-for-contig-coverage-depth">Calculate summary statistics for contig coverage depth</h4>
<ul>
<li>Go to <ss>Tools &rarr; Statistics and Visualisation &rarr; Statistics &rarr; Summary Statisitics</ss> and set the following:  </li>
<li><ss>Summary statistics on</ss>: <fn>Remove beginning on Data 4</fn></li>
<li><ss>Column or expression</ss>: <em>c2</em></li>
<li>Click <ss>Execute</ss></li>
<li>You’ll note that the summary statistics tool failed (red background in the <ss>History</ss>). There was an error!</li>
<li>If you click on the filename, and then the bug symbol <img alt="bug icon" src="./images/image08.png" />, it will tell you what went wrong. (There is a missing python library).</li>
<li>At this point, you would normally contact your Galaxy server administrator.</li>
</ul>
<h3 id="convert-file-formats">Convert file formats</h3>
<p>This shows how to convert a FASTQ file to a FASTA file. The tool creates a new file with the converted data.</p>
<ul>
<li>Go to <ss>Tools &rarr; Basic Tools &rarr; Convert Formats &rarr; FASTQ to FASTA</ss> and set the following:</li>
<li><ss>FASTQ file to convert</ss>: <fn>typical.fastq</fn></li>
<li>Click <ss>Execute</ss></li>
<li>The output is a new Fasta file called <fn>FASTQ to FASTA on data 2</fn>.</li>
</ul>
<h3 id="find-features">Find features</h3>
<p>This example shows how to use a tool called “barrnap” to search for rRNAs in a DNA sequence.</p>
<h4 id="find-all-of-the-ribosomal-rnas-in-a-sequence">Find all of the ribosomal RNAs in a sequence</h4>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Annotation &rarr; barrnap</ss> and set the following:</li>
<li><ss>Fasta file</ss>: <fn>MRSA252.fna</fn></li>
<li>Click <ss>Execute</ss></li>
<li>The output is <fn>barrnap on data 3</fn> It is a gff3 format file (general feature format version 3). Each line in the file describes a feature in the DNA sequence.</li>
</ul>
<h4 id="filter-the-annotations-to-get-the-23s-rnas">Filter the annotations to get the 23S RNAs</h4>
<ul>
<li>Make a file with only the 23S rRNA features</li>
<li>Go to <ss>Tools &rarr; Basic Tools &rarr; Filter and Sort &rarr; Select</ss> and set the following:</li>
<li><ss>Select lines from</ss>: (whatever you called the barrnap gff3 output)</li>
<li><ss>the pattern</ss>: <em>23S</em> (this will look for all the lines in the file that contain “23S”)</li>
<li>Click <ss>Execute</ss></li>
<li>Now you have a gff3 file with just the 23S annotations!</li>
</ul>
<h2 id="what-next">What next?</h2>
<ul>
<li>
<p>Remember how we started a new <ss>History</ss> at the beginning? If you want to see any of your old histories, click on the History menu button <img alt="history button" src="./images/image02.png" /> at the top of the <ss>History</ss> panel and then select “Saved Histories.” This will give you a list of all the histories you have worked on in this Galaxy server.</p>
</li>
<li>
<p><a href="../data-dna/">Upload a saved Galaxy history (a set of files) to use in these tutorials</a>.</p>
</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>