
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
        <link rel="canonical" href="http://sepsis-omics.github.io/tutorials/modules/snippy/">
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <link rel="shortcut icon" href="../../favicon.ico">
      
      <meta name="generator" content="mkdocs-0.17.2, mkdocs-material-1.6.1">
    
    
      
        <title>Variant finding - ABRPI-Training</title>
      
    
    
      <script src="../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application-4d0d3f2fbf.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
  
  
  
    <body>
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="http://sepsis-omics.github.io/tutorials/" title="ABRPI-Training" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Genomics
                </span>
              
            
            Variant finding
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
            


  


  <a href="https://github.com/sepsis-omics/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../workshop_overview/" title="Genomics" class="md-tabs__link md-tabs__link--active">
          Genomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../dge/" title="Transcriptomics" class="md-tabs__link">
          Transcriptomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xtandem/" title="Proteomics" class="md-tabs__link">
          Proteomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xcms/" title="Metabolomics" class="md-tabs__link">
          Metabolomics
        </a>
      
    </li>
  

      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    ABRPI-Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/sepsis-omics/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../workshop_overview/" title="Genomics workshop overview" class="md-nav__link">
      Genomics workshop overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pear/" title="Merge reads with Pear" class="md-nav__link">
      Merge reads with Pear
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../velvet/" title="Genome assembly with Velvet" class="md-nav__link">
      Genome assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly with Spades" class="md-nav__link">
      Genome assembly with Spades
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Variant finding
      </label>
    
    <a href="./" title="Variant finding" class="md-nav__link md-nav__link--active">
      Variant finding
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-reference" title="Prepare reference" class="md-nav__link">
    Prepare reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#call-variants-with-snippy" title="Call variants with Snippy" class="md-nav__link">
    Call variants with Snippy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-snippy-output" title="Examine Snippy output" class="md-nav__link">
    Examine Snippy output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#view-snippy-output-in-jbrowse" title="View Snippy output in JBrowse" class="md-nav__link">
    View Snippy output in JBrowse
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../access_data/" title="Public data &rarr; assembly, annotation" class="md-nav__link">
      Public data &rarr; assembly, annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../abricate/" title="Finding antibiotic-resistant genes" class="md-nav__link">
      Finding antibiotic-resistant genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../viral_genomes/" title="Viral genome sequencing" class="md-nav__link">
      Viral genome sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows" class="md-nav__link">
      Galaxy workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../frogs/" title="Metagenomics with FROGS" class="md-nav__link">
      Metagenomics with FROGS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification with Kraken" class="md-nav__link">
      Sample identification with Kraken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pacbio/" title="PacBio assembly with SMRT portal" class="md-nav__link">
      PacBio assembly with SMRT portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cmdline_assembly/" title="PacBio assembly with command line tools" class="md-nav__link">
      PacBio assembly with command line tools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades_cmdline/" title="Spades assembly - commandline" class="md-nav__link">
      Spades assembly - commandline
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mauve/" title="Genome alignment with Mauve" class="md-nav__link">
      Genome alignment with Mauve
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary/" title="Pangenomes with Roary/Phandango - command line" class="md-nav__link">
      Pangenomes with Roary/Phandango - command line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary-galaxy/" title="Pangenomes with Roary/Phandango - Galaxy" class="md-nav__link">
      Pangenomes with Roary/Phandango - Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cmdline_assembly_v2/" title="Long read assembly workshop" class="md-nav__link">
      Long read assembly workshop
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../dge/" title="Differential gene expression using Galaxy and Degust" class="md-nav__link">
      Differential gene expression using Galaxy and Degust
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kallisto/" title="Differential gene expression using Kallisto and Degust" class="md-nav__link">
      Differential gene expression using Kallisto and Degust
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xtandem/" title="Protein Identification" class="md-nav__link">
      Protein Identification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pathway_tools/annotation/" title="Pathway annotation and prediction" class="md-nav__link">
      Pathway annotation and prediction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Metabolomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metabolomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xcms/" title="Metabolite identification" class="md-nav__link">
      Metabolite identification
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-reference" title="Prepare reference" class="md-nav__link">
    Prepare reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#call-variants-with-snippy" title="Call variants with Snippy" class="md-nav__link">
    Call variants with Snippy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-snippy-output" title="Examine Snippy output" class="md-nav__link">
    Examine Snippy output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#view-snippy-output-in-jbrowse" title="View Snippy output in JBrowse" class="md-nav__link">
    View Snippy output in JBrowse
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/sepsis-omics/tutorials/edit/master/docs/modules/snippy/index.md" title="Edit this page" class="md-icon md-content__icon">edit</a>
                
                
                <p><br></p>
<h1 id="variant-calling-with-snippy">Variant calling with Snippy</h1>
<p>Keywords: variant calling, SNP, Snippy, JBrowse, Galaxy, Microbial Genomics Virtual Lab</p>
<h2 id="background">Background</h2>
<p>Variant calling is the process of identifying differences between two genome samples.
Usually differences are limited to single nucleotide polymorphisms (SNPs) and small insertions and deletions (indels). Larger structural variation such as inversions, duplications and large deletions are not typically covered by &ldquo;variant calling&rdquo;.</p>
<p>In this tutorial, we will use the tool &ldquo;Snippy&rdquo; (link to Snippy is <a href="https://github.com/tseemann/snippy">here</a>). Snippy uses a tool to align the reads to a reference genome, and another tool to decide (&ldquo;call&rdquo;) if the discrepancies are real variants.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<ol>
<li>Find variants between a reference genome and a set of reads</li>
<li>Visualise the SNP in context of the reads aligned to the genome</li>
<li>Determine the effect of those variants on genomic features</li>
<li>Understand if the SNP is potentially affecting the phenotype</li>
</ol>
<h2 id="prepare-reference">Prepare reference</h2>
<!-- We will import  history from Galaxy:

- In the menu options across the top, go to <ss>Shared Data</ss>.
- Click on <ss>Histories</ss>.
- A list of published histories should appear. Click on <fn>GCC 2016 small genome variants</fn>.
- Click on <ss>Import history</ss>.
- An option will appear to re-name the history. We don't need to rename it, so click <ss>Import</ss>.
- The history will now appear in your Current History pane, and the files are now ready to use in Galaxy analyses.
-->

<!-- We will use the same data that we used in the [Assembly with Spades tutorial.](../spades/index.md) This should still be in your current galaxy history. If not, re-import the data into a new history using the instructions in that tutorial.-->

<p>For variant calling, we need a reference genome that is of the same strain as the input sequence reads.</p>
<p>For this tutorial, our reference is the <fn>wildtype.gbk</fn> file and our reads are <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>. </p>
<p>If these files are not presently in your Galaxy history, import them from the <a href="../data-dna/">Training dataset page.</a></p>
<h2 id="call-variants-with-snippy">Call variants with Snippy</h2>
<ul>
<li>Go to the Galaxy tools panel, and use the search box at the top to search for &ldquo;snippy&rdquo;. </li>
<li>For <ss>Reference type</ss> select <em>Genbank</em>.</li>
<li>Then for <ss>Reference Genbank</ss> choose the <fn>wildtype.gbk</fn> file.</li>
<li>For <ss>Single or Paired-end reads</ss> choose <em>Paired</em>.</li>
<li>Then choose the first set of reads, <fn>mutant_R1.fastq</fn> and second set of reads, <fn>mutant_R2.fastq</fn>.</li>
<li>For <ss>Cleanup the non-snp output files</ss> select <em>No</em>.</li>
</ul>
<p>Your tool interface should look like this:</p>
<p><img alt="Snippy interface" src="./images/interface.png" /></p>
<ul>
<li>Click <ss>Execute</ss>.</li>
</ul>
<h2 id="examine-snippy-output">Examine Snippy output</h2>
<!-- First, enable "Scratchbook" in Galaxy - this allows you to view several windows simultaneously. Click on the squares:

![scratchbook icon](images/scratchbook.png)

-->

<p>From Snippy, there are 10 output files in various formats.</p>
<ul>
<li>Go to the file called <fn>snippy on data XX, data XX and data XX table</fn> and click on the eye icon.</li>
<li>We can see a list of variants. Look in column 3 to see which types the variants are, such as a SNP or a deletion.</li>
<li>Look at the third variant called. This is a T&rarr;A mutation, causing a stop codon. Look at column 14: the product of this gene is a methicillin resistance protein. Methicillin is an antibiotic. What might be the result of such a mutation? <!--[add a hint/info box]--></li>
</ul>
<h2 id="view-snippy-output-in-jbrowse">View Snippy output in JBrowse</h2>
<ul>
<li>
<p>Go to the Galaxy tools panel, and use the search box at the top to search for &ldquo;JBrowse&rdquo;. </p>
</li>
<li>
<p>Under <ss>Reference genome to display</ss> choose <em>Use a genome from history</em>.</p>
</li>
<li>
<p>Under <ss>Select the reference genome</ss> choose <fn>wildtype.fna</fn>. This sequence will be the reference against which annotations are displayed.</p>
</li>
<li>
<p>For <ss>Produce a Standalone Instance</ss> select <em>Yes</em>.</p>
</li>
<li>
<p>For <ss>Genetic Code</ss> choose <em>11: The Bacterial, Archaeal and Plant Plastid Code</em>.</p>
</li>
<li>
<p>Under <ss>JBrowse-in-Galaxy Action</ss> choose <em>New JBrowse Instance</em>.</p>
</li>
<li>
<p>We will now set up three different tracks - these are datasets displayed underneath the reference sequence (which is displayed as nucleotides in FASTA format). We will choose to display the sequence reads (the .bam file), the variants found by snippy (the .gff file) and the annotated reference genome (the wildtype.gff)</p>
</li>
</ul>
<p><em>Track 1 - sequence reads</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss></li>
<li>For <ss>Track Cateogry</ss> name it &ldquo;sequence reads&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>BAM Pileups</em></li>
<li>For <ss>BAM Track Data</ss> select <fn>the snippy bam file</fn></li>
<li>For <ss>Autogenerate SNP Track</ss> select <em>Yes</em></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 2 - variants</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss>Track Category</ss> name it &ldquo;variants&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>the snippy snps gff file</fn></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 3 - annotated reference</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss> Track Category</ss> name it &ldquo;annotated reference&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>wildtype.gff</fn></li>
<li>Under <ss>JBrowse Track Type[Advanced]</ss> select <em>Canvas Features</em>.</li>
<li>Click on <ss>JBrowse Styling Options <Advanced]</ss></li>
<li>Under <ss>JBrowse style.label</ss> add in the word <em>product</em>.</li>
<li>Under <ss>JBrowse style.description</ss> add in the word <em>product</em>.</li>
<li>
<p>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</p>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
<li>
<p>A new file will be created, called <fn>JBrowse on data XX and data XX - Complete</fn>. Click on the eye icon next to the file name. The JBrowse window will appear in the centre Galaxy panel.</p>
</li>
<li>
<p>On the left, tick boxes to display the tracks</p>
</li>
<li>
<p>Use the minus button to zoom out to see:</p>
<ul>
<li>sequence reads and their coverage (the grey graph)</li>
</ul>
</li>
<li>
<p>Use the plus button to zoom in to see:</p>
<ul>
<li>probable real variants (a whole column of snps)</li>
<li>probable errors (single one here and there)</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse1.png" /></p>
<ul>
<li>In the coordinates box, type in <em>47299</em> and then <ss>Go</ss> to see the position of the SNP discussed above.<ul>
<li>the correct codon at this position is TGT, coding for the amino acid Cysteine, in the middle row of the amino acid translations.</li>
<li>the mutation of T &rarr; A turns this triplet into TGA, a stop codon.</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse2.png" />    </p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../prokka/" title="Genome annotation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Genome annotation
              </span>
            </div>
          </a>
        
        
          <a href="../access_data/" title="Public data &rarr; assembly, annotation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Public data &rarr; assembly, annotation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2016-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://omics.data.edu.au/" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application-6b599127bc.js"></script>
      <script>app.initialize({url:{base:"../.."}})</script>
      
        <script src="../../search/require.js"></script>
      
        <script src="../../search/search.js"></script>
      
    
    
      
      <script>!function(e,t,a,n,o,c,i){e.GoogleAnalyticsObject=o,e[o]=e[o]||function(){(e[o].q=e[o].q||[]).push(arguments)},e[o].l=1*new Date,c=t.createElement(a),i=t.getElementsByTagName(a)[0],c.async=1,c.src=n,i.parentNode.insertBefore(c,i)}(window,document,"script","https://www.google-analytics.com/analytics.js","ga"),ga("create","UA-21794901-5","www.melbournebioinformatics.org.au/tutorials/"),ga("set","anonymizeIp",!0),ga("send","pageview");var links=document.getElementsByTagName("a");Array.prototype.map.call(links,function(e){e.host!=document.location.host&&e.addEventListener("click",function(){var t=e.getAttribute("data-md-action")||"follow";ga("send","event","outbound",t,e.href)})});var query=document.forms.search.query;query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})</script>
      
    
  </body>
</html>