<br>
# Welcome!

<ss> Update June 2024: </ss>
<br>
<ss>These tutorials were initially developed for the [Antibiotic Resistant Pathogens Initiative](https://bioplatforms.com/projects/sepsis/) </ss>
<br>
<ss>
Galaxy-based tutorials have now been merged into the [Galaxy Training Network](https://training.galaxyproject.org/training-material/) </ss>
<br>
<ss> Please check there for the most up-to-date material, as this page is no longer maintained </ss>


## ABRPI training materials

### Antibiotic Resistant Pathogens Initiative


- This site contains tutorials for using the analysis components of the [omics.data.edu.au](omics.data.edu.au) platform to perform various bioinformatics tasks on bacterial “omics” data.


- Tools covered by these tutorials include:

    - those within the [Microbial Genomics Virtual Lab](http://genome.edu.au/) (the mGVL), which can be accessed either through the [hosted ABPRI-Galaxy service](http://abrpi.genome.edu.au/galaxy/), or by setting up your own mGVL instance (see below) and accessing via the Unix command line.

    - those within the [hosted ABPRI-PathwayTools service](http://abrpi.genome.edu.au:9655).


- Tutorials are listed under the tabs in the top panel (Genomics, Transcriptomics, Proteomics, Metabolomics).
_______

- If you wish to set up your own instance (version) of the mGVL, follow the instructions [here](http://vlsci.github.io/lscc_docs/tutorials/gvl_launch/gvl_launch/).


    - Note: at the stage where you select options in the GVL Launcher window, go to *Show advanced startup options* and under Flavor select *Microbial GVL with Tutorial Indices*. In this mGVL instance, you can use both Galaxy and command line tools. If you wish to use the command line tools on a different computer (e.g. your local computer), you would need to make sure the required tools are installed (e.g. Canu, Circlator, Pilon, SPAdes, etc.).

- For additional tutorials about how to use the Genomics Virtual Lab and Galaxy, see <https://www.gvl.org.au/learn/>




<!--

- The Food and Health Flagship is an [RDS-funded](http://omics.data.edu.au/) project to provide cloud-based data services and tools
for Australian Life Science Researchers to combine, analyse and interpret
genomic, transcriptomic, proteomic and metabolomic data. The data platform will incorporate the [Bioplatforms Australia Antibiotic Resistant Pathogens Initiative (ABRPI)](http://www.bioplatforms.com/antibiotic-resistant-pathogens/).


- This site contains tutorials for using the ABRPI
[Microbial Genomics Virtual Lab](http://genome.edu.au/) (the mGVL) to perform bioinformatics
tasks on bacterial "omics" data, either on the Unix command line or using
the [Galaxy](http://galaxyproject.org/) system.

- Tutorials are listed under the tabs in the top panel and are being added progressively.

- If you wish to set up your own instance (version) of the mGVL, follow the instructions [here](http://vlsci.github.io/lscc_docs/tutorials/gvl_launch/gvl_launch/).
    - Note: at the stage where you select options in the GVL Launcher window, go to <fn>Show advanced startup options</fn> and under <fn>Flavor</fn> select *Microbial GVL with Tutorial Indices*. In this mGVL instance, you can use both Galaxy and command line tools. If you wish to use the command line tools on a different computer (e.g. your local computer), you would need to make sure the required tools are installed (e.g. Canu, Circlator, Pilon, SPAdes, etc.).

- For more information about how to use the Microbial Genomics Virtual Lab and Galaxy, see <http://www.genome.edu.au/>

-->


![sepsis omics logo](media/logos/omics.png)
