<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Torsten Seemann, Anna Syme">
        
        <link rel="shortcut icon" href="../../../favicon.ico">
        

	<title>Assemble with Spades - Sepsis Tutorials</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">Sepsis Tutorials</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Introductory training <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../overview/">Overview</a>
</li>

                    
                        
<li >
    <a href="../1a/">Use Galaxy</a>
</li>

                    
                        
<li >
    <a href="../1c/">View with JBroswe</a>
</li>

                    
                        
<li class="active">
    <a href="./">Assemble with Spades</a>
</li>

                    
                        
<li >
    <a href="../2b/">Annotate with Prokka</a>
</li>

                    
                        
<li >
    <a href="../3b/">Find variants with Snippy</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Advanced training <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../how_to/overview/">Overview</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../../tools/">Oveview</a>
</li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Genomics</a>
    <ul class="dropdown-menu">
        
            
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Quality Control</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../../dna/qualitycontrol/fastqc/">FastQC in Galaxy</a>
</li>

        
            
<li >
    <a href="../../../dna/qualitycontrol/fastqc_cmdline/">FastQC on commandline</a>
</li>

        
            
<li >
    <a href="../../../dna/qualitycontrol/trimmomatic/">Trimmomatic in Galaxy</a>
</li>

        
    </ul>
  </li>

        
    </ul>
  </li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../1c/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../2b/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#assembly-using-spades">Assembly using Spades</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning objectives</a></li>
        
            <li><a href="#import-data-into-galaxy">Import data into Galaxy</a></li>
        
            <li><a href="#view-files">View files</a></li>
        
            <li><a href="#evaluate-the-input-reads">Evaluate the input reads</a></li>
        
            <li><a href="#assemble-reads-with-spades">Assemble reads with Spades</a></li>
        
            <li><a href="#examine-the-output">Examine the output</a></li>
        
            <li><a href="#what-next">What next?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="assembly-using-spades">Assembly using Spades</h1>
<h2 id="background">Background</h2>
<p>Spades is one of a number of <em>de novo</em> assemblers that use short read sets as input (e.g. Illumina Reads), and the assembly method is based on de Bruijn graphs. For information about Spades see this <a href="http://bioinf.spbau.ru/spades">link</a>.</p>
<!---
A protocol for assembling with Velvet (another *de novo* assembler) is available [here](https://docs.google.com/document/d/1xs-TI5MejQARqo0pcocGlymsXldwJbJII890gnmjI0o/pub).
--->

<p>In this activity, we will perform a <em>de novo</em> assembly of a short read set using the Spades assembler. The output from Spades that we are interested in is a multifasta file that contains the draft genome sequence.</p>
<p>The read set for today is from an imaginary <em>Staphylococcus aureus</em> bacterium with a miniature genome.</p>
<p>We have a closed, annotated genome sequence for a closely related <em>wildtype</em> strain.</p>
<p>The whole genome shotgun method used to sequence our mutant strain read set was produced on an Illumina DNA sequencing instrument.</p>
<ul>
<li>The reads are paired-end</li>
<li>Each read is 150 bases (before trimming)</li>
<li>The number of bases sequenced is equivalent to 19x the genome sequence of the wildtype strain. (Read coverage 19x - rather low!).</li>
</ul>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>import data into Galaxy  </li>
<li>view the files</li>
<li>evaluate the read quality</li>
<li>assemble the reads using Spades, and</li>
<li>examine the output assembly.</li>
</ol>
<h2 id="import-data-into-galaxy">Import data into Galaxy</h2>
<ul>
<li>Go to the address of your galaxy server.</li>
</ul>
<!---
- [Remind me how to logon.](https://docs.google.com/document/d/1LAQvhIG8s-vv6T14bb8lGRkmoNha7E3bHf9kAgUwMs0/pub)
FIXME: note this contains the same galaxy address as above - change?
--->

<ul>
<li>Click on the <ss>Analyze Data</ss> menu at the top of the page.    </li>
<li>Click on the <ss>History options</ss> button the <img alt="history button" src="./images/image02.png" /> on the top right of the history pane.</li>
<li>Click <ss>Import from File</ss> (at the bottom of the list).  </li>
<li>A new page will appear with a text box for the URL of the history to import.  </li>
<li>Copy the following URL into the text box: <a href="http://43.240.98.1/public/dieter/Galaxy-History-Colombiaworkshopstart.tar.gz">http://43.240.98.1/public/dieter/Galaxy-History-Colombiaworkshopstart.tar.gz</a>  </li>
<li>Click <ss>Submit</ss>.  </li>
<li>Galaxy will download the data files from the internet and will be available as an additional history (takes about one minute).  </li>
<li>To view this new history, click the <ss>View all histories</ss> button<img alt="Histories button" src="./images/image01.png" /> (top right of the history pane).  </li>
<li>If the history has finished downloading it will appear as <fn>imported from archive: Colombia_workshop_start</fn></li>
<li>Click on the <img alt="Switch button" src="./images/image06.png" /> button above the <fn>imported from archive:Colombia_workshop_start</fn> then the <img alt="Done button" src="./images/image05.png" /> button.</li>
<li>You should now have four files in the history pane as follows:</li>
</ul>
<p><img alt="Files in history" src="./images/image07.png" /></p>
<h2 id="view-files">View files</h2>
<p>All the files are text files.</p>
<ul>
<li><fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>: a paired-end read set  </li>
<li><fn>wildtype.fna</fn>: a file that contains the genome sequence of the wildtype strain in fasta format (a header line, then the nucleotide sequence of the genome)</li>
<li><fn>wildtype.gff</fn>: a file that contains the genome sequence of the wildtype strain in general feature format. (a list of features - one feature per line, then the nucleotide sequence of the genome)</li>
</ul>
<p>Look at the contents of these files.</p>
<ul>
<li>Click on the View Data button (the <img alt="Eye icon" src="./images/image04.png" />) next to each of the files in turn.</li>
<li>The gff file should look like this:</li>
</ul>
<!---
- Brief Discussion about the GFF format (FIXME: add)
--->

<p><img alt="GFF format" src="./images/image08.png" /></p>
<h2 id="evaluate-the-input-reads">Evaluate the input reads</h2>
<p>Questions you might ask about your input reads include:</p>
<ul>
<li>How good is my read set?</li>
<li>Do I need to ask for a new sequencing run?  </li>
<li>Is it suitable for the analysis I need to do?</li>
</ul>
<p>We will evaluate the input reads using the FastQC tool.</p>
<ul>
<li>This runs a standard series of tests on your read set and returns a relatively easy-to-interpret report.</li>
<li>We will use the FastQC tool in Galaxy to evaluate the quality of one of our FASTQ files.</li>
<li>Go to <ss>Tools &rarr; NGS:Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss></li>
<li>Select <fn>mutant_R1.fastq</fn></li>
<li><ss>Execute</ss></li>
<li>Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint:<img alt="Eye icon" src="./images/image04.png" />). It has a summary at the top of
the page and a number of graphs.</li>
</ul>
<p>Some of the important outputs of FastQC for our purposes are:</p>
<ul>
<li><ss>Basic Statistics: Sequence length</ss>: will be important in setting maximum k-mer size value for assembly</li>
<li><ss>Basic Statistics: Encoding</ss>: Quality encoding type: important for quality trimming software</li>
<li><ss>Basic Statistics: % GC</ss>: high GC organisms don’t tend to assemble well and may have an uneven read coverage distribution.</li>
<li><ss>Basic Statistics: Total sequences</ss>: Total number of reads: gives you an idea of coverage.</li>
<li><ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run.</li>
<li><ss>Per base N content</ss>: Presence of large numbers of Ns in reads: may point to poor quality sequencing run. You would need to trim these reads to remove Ns.</li>
<li><ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes or adapter sequences.</li>
</ul>
<p>Although we have warnings for two outputs (per base sequence content; Kmer content), we can ignore these for now. For a fuller discussion of FastQC outputs and warnings, see the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC website link</a>, including the section on each of the output <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">reports</a>, and examples of <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">&ldquo;good&rdquo;</a> and <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">&ldquo;bad&rdquo;</a> Illumina data. We won’t be doing anything to these data to clean it up as there isn’t much need. Therefore we will get on with the assembly!</p>
<h2 id="assemble-reads-with-spades">Assemble reads with Spades</h2>
<ul>
<li>We will perform a <em>de novo</em> assembly of the mutant FASTQ reads into long contiguous sequences (in FASTA format.)</li>
</ul>
<!---
- Spades produces both contigs and scaffolds.
Ask your demonstrator if you would like to know the difference between contigs and scaffolds.
--->

<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; spades</ss></li>
<li>
<p>Set the following parameters:</p>
<ul>
<li><ss>Run only Assembly</ss>: <em>Yes</em>  </li>
<li><ss>Kmers to use separated by commas:</ss> <em>33,55,91</em>  [no spaces]  </li>
<li><ss>Coverage cutoff:</ss> <em>auto</em>  </li>
<li><ss>Files &rarr; Forward reads:</ss> <fn>mutant_R1.fastq</fn>  </li>
<li><ss>Files &rarr; Reverse reads:</ss> <fn>mutant_R2.fastq</fn>  </li>
</ul>
</li>
<li>
<p>Your tool interface should look like this:</p>
</li>
</ul>
<p><img alt="Spades interface" src="./images/image03.png" /></p>
<ul>
<li>Click <ss>Execute</ss></li>
</ul>
<h2 id="examine-the-output">Examine the output</h2>
<ul>
<li>Galaxy is now running Spades on the reads for you.</li>
<li>
<p>When it is finished, you will have five new files in your history.  </p>
<ul>
<li>two FASTA files of the resulting contigs and scaffolds</li>
<li>two files for statistics about these</li>
<li>the Spades logfile</li>
</ul>
</li>
<li>
<p>Click on the View Data button <img alt="Eye icon" src="./images/image04.png" /> on each of the files.</p>
</li>
<li>Note that the short reads have been assembled into much longer contigs.</li>
<li>(However, in this case, the contigs have not been assembled into larger scaffolds.)</li>
<li>The stats files will give you the length of each of the contigs.</li>
</ul>
<h2 id="what-next">What next?</h2>
<ul>
<li><a href="../2b/">Annotate the genome using Prokka.</a></li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>